### PR TITLE
LPS-180271 No text in repetable fields group's field topbar when upgrading from 7.3.10.1

### PR DESCRIPTION
**Pull request by Mikellorza**
Bug : https://issues.liferay.com/browse/LPS-180271

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java [hunk 1]
```diff
@@ -272,19 +272,24 @@ private DDMFormField _createFieldSetDDMFormField(
...
 				setProperty("collapsible", false);
 				setProperty("ddmStructureId", ddmStructureId);
 				setProperty("ddmStructureLayoutId", ddmStructureLayoutId);
+				setProperty("labelAtStructureLevel", true);
 				setProperty("upgradedStructure", upgradedStructure);
 				setReadOnly(false);
 				setRepeatable(repeatable);
 				setRequired(false);
-				setShowLabel(false);
+				setShowLabel(true);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java:292]: @Mikellorza I believe the reported failures [here](https://github.com/liferay-forms/liferay-portal/pull/2216#issuecomment-1544098218) ([DDMDataDefinitionConverterTest](https://test-1-34.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/485659/testReport/com.liferay.dynamic.data.mapping.util.test/DDMDataDefinitionConverterTest)) may be related to these changes.

Would you mind taking a look?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+						{
+							for (Locale locale : availableLocales) {
+								addString(
+									locale,
+									_language.get(locale, "fields-group"));
+							}
+						}
+					});
+				ddmFormField.setProperty("labelAtStructureLevel", true);
+				ddmFormField.setShowLabel(true);
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java:82]: Yes, right, thanks for remind it

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+			}
+		}
+
+		return false;
+	}
+
+	private void _upgradeStructureDefinition() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				"select * from DDMStructure where classNameId = ? or " +
+					"classNameId = ? order by createDate");
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java:121]: ok, no problem

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructure set parentStructureId = 0, " +
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java:125]: it is no needed, thanks

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+				preparedStatement2.executeBatch();
+			}
+		}
+	}
+
+	private void _upgradeStructureVersionDefinition() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				StringBundler.concat(
+					"select DDMStructure.structureKey, DDMStructureVersion.* ",
+					"from DDMStructureVersion inner join DDMStructure on ",
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java:155]: ok, no problem

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+					"DDMStructure.structureId = ",
+					"DDMStructureVersion.structureId where ",
+					"DDMStructure.classNameId = ? or DDMStructure.classNameId ",
+					"= ?"));
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructureVersion set parentStructureId = 0, " +
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java:163]: it is no needed here, thanks

**Comment by Mikellorza** [comment]
ci:test:sf

**Comment by Mikellorza** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java (hunk 1 above)
Comment (Mikellorza) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java:302]: fixed, thanks

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+	private boolean _isLabelSet(DDMFormField ddmFormField) {
+		LocalizedValue localizedValue = ddmFormField.getLabel();
+
+		Map<Locale, String> values = localizedValue.getValues();
+
+		for (Map.Entry<Locale, String> value : values.entrySet()) {
+			if (Validator.isNotNull(value.getValue())) {
+				return true;
+			}
+		}
```
[the remark this answers is not in the document]
Comment (Mikellorza) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_1/DDMStructureUpgradeProcess.java]: Ok, no problem

**Comment by Mikellorza** [comment]
@carolmariaabb everything you proposed it is now implemented. 

Thanks!

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Pull request closed without merging**